core/src/main/java/com/amplifyframework/analytics/UserProfile.java

```diff
@@ -0,0 +1,313 @@
+                return false;
+            if (postalCode != null ? !postalCode.equals(location.postalCode) : location.postalCode != null)
+                return false;
+            if (city != null ? !city.equals(location.city) : location.city != null) return false;
+            if (region != null ? !region.equals(location.region) : location.region != null)
+                return false;
+            return country != null ? country.equals(location.country) : location.country == null;
+        }
+
+        @SuppressWarnings("checkstyle:MagicNumber")
+        @Override
+        public int hashCode() {
+            int result = latitude != null ? latitude.hashCode() : 0;
+            result = 31 * result + (longitude != null ? longitude.hashCode() : 0);
+            result = 31 * result + (postalCode != null ? postalCode.hashCode() : 0);
+            result = 31 * result + (city != null ? city.hashCode() : 0);
+            result = 31 * result + (region != null ? region.hashCode() : 0);
+            result = 31 * result + (country != null ? country.hashCode() : 0);
+            return result;
+        }
+
+        @Override
+        public String toString() {
+            return "Location{" +
+                    "latitude=" + latitude +
+                    ", longitude=" + longitude +
+                    ", postalCode='" + postalCode + '\'' +
+                    ", city='" + city + '\'' +
+                    ", region='" + region + '\'' +
+                    ", country='" + country + '\'' +
+                    '}';
+        }
+
+        public static final class Builder {
+            private Double latitude;
+            private Double longitude;
+            private String postalCode;
+            private String city;
+            private String region;
+            private String country;
+
+            public Builder latitude(@NonNull Double latitude) {
+                Objects.requireNonNull(latitude);
+                this.latitude = latitude;
+                return this;
+            }
+
+            public Builder longitude(@NonNull Double longitude) {
+                Objects.requireNonNull(longitude);
+                this.longitude = longitude;
+                return this;
+            }
+
+            public Builder postalCode(@NonNull String postalCode) {
+                Objects.requireNonNull(postalCode);
+                this.postalCode = postalCode;
+                return this;
+            }
+
+            public Builder city(@NonNull String city) {
+                Objects.requireNonNull(city);
+                this.city = city;
+                return this;
+            }
+
+            public Builder region(@NonNull String region) {
+                Objects.requireNonNull(region);
+                this.region = region;
+                return this;
+            }
+
+            public Builder country(@NonNull String country) {
```

Comment:
All of these things that return a `Builder` will all return `@NonNull`, too. Might as well add it everywhere you can on public/protected members. It makes a big help for Kotlin code that calls this stuff.